Support MacOS

The build for MacOS currently fails.  After configuration with
```
./configure --prefix=$prefix --host=$target --disable-dependency-tracking
```
`make` gives the following error:
```
libtool: link: /opt/x86_64-linux-gnu/tools/clang -target x86_64-apple-darwin14 --sysroot /opt/x86_64-apple-darwin14/x86_64-apple-darwin14/sys-root -mmacosx-version-min=10.8 -Wl,-syslibroot -Wl,/opt/x86_64-apple-darwin14/x86_64-apple-darwin14/sys-root -o siman_tsp siman_tsp.o  ./.libs/libgslsiman.a ../rng/.libs/libgslrng.a ../ieee-utils/.libs/libgslieeeutils.a ../err/.libs/libgslerr.a ../sys/.libs/libgslsys.a ../utils/.libs/libutils.a -lm
ld: warning: ignoring file ../err/.libs/libgslerr.a, file was built for archive which is not the architecture being linked (x86_64): ../err/.libs/libgslerr.a
Undefined symbols for architecture x86_64:
  "_gsl_error", referenced from:
      _gsl_rng_alloc in libgslrng.a(rng.o)
      _gsl_rng_memcpy in libgslrng.a(rng.o)
      _gsl_rng_clone in libgslrng.a(rng.o)
      _gsl_rng_env_setup in libgslrng.a(default.o)
      _gsl_rng_uniform_int in libgslrng.a(inline.o)
      _gsl_ieee_read_mode_string in libgslieeeutils.a(read.o)
      _gsl_ieee_set_mode in libgslieeeutils.a(fp.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang-6.0: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [Makefile:605: siman_tsp] Error 1
make[2]: Leaving directory '/workspace/srcdir/gsl-1.16/siman'
make[1]: *** [Makefile:955: all-recursive] Error 1
make[1]: Leaving directory '/workspace/srcdir/gsl-1.16'
make: *** [Makefile:657: all] Error 2
```
After reading similar issues on the Internet, I think that there may be some problems with the `ar` tool used.  `AR` is set to:
```
sandbox:$WORKSPACE/srcdir/gsl-1.16# echo $AR
/opt/x86_64-linux-gnu/tools/llvm-ar
```
Any help to fix this issue would be much appreciated

--- a/.travis.yml
+++ b/.travis.yml
@@ -14,7 +14,7 @@ cache:
 env:
   global:
     - BINARYBUILDER_DOWNLOADS_CACHE=downloads
-    - BINARYBUILDER_AUTOMATIC_APPLE=false
+    - BINARYBUILDER_AUTOMATIC_APPLE=true
 sudo: required
 
 # Before anything else, get the latest versions of things

--- a/build_tarballs.jl
+++ b/build_tarballs.jl
@@ -19,7 +19,6 @@ cd $WORKSPACE/srcdir/gsl-*/
 ./configure --prefix=$prefix --host=${target}
 make -j${nproc}
 make install
-
 """
 
 # These are the platforms we will build for by default, unless further

--- a/build_tarballs.jl
+++ b/build_tarballs.jl
@@ -14,6 +14,7 @@ sources = [
 
 # Bash recipe for building across all platforms
 script = raw"""
+AR=/opt/${target}/bin/ar
 cd $WORKSPACE/srcdir/gsl-*/
 ./configure --prefix=$prefix --host=${target}
 make -j${nproc}
@@ -23,20 +24,7 @@ make install
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = [
-    Linux(:i686, libc=:glibc),
-    Linux(:x86_64, libc=:glibc),
-    Linux(:aarch64, libc=:glibc),
-    Linux(:armv7l, libc=:glibc, call_abi=:eabihf),
-    Linux(:powerpc64le, libc=:glibc),
-    Linux(:i686, libc=:musl),
-    Linux(:x86_64, libc=:musl),
-    Linux(:aarch64, libc=:musl),
-    Linux(:armv7l, libc=:musl, call_abi=:eabihf),
-    FreeBSD(:x86_64),
-    Windows(:i686),
-    Windows(:x86_64)
-]
+platforms = supported_platforms()
 
 # The products that we will ensure are always built
 products(prefix) = [

